src/cpp/src/visual_language/qwen2vl/classes.cpp

```diff
@@ -292,6 +301,125 @@ ov::Tensor merge_text_and_image_embeddings(
+    last_shape->output(0).get_tensor().set_names({"last_shape"});
+
+    // res.buf[i] = ((float(v2) / 255.0f) - m3[c]) / s3[c];
+    auto raw_images_planar = std::make_shared<ov::op::v1::Transpose>(raw_images, 
+        std::make_shared<ov::op::v0::Constant>(element::i32, Shape{4}, std::vector<int32_t>{0, 3, 1, 2})
+    );
+    ov::op::v0::Interpolate::Attributes attrs = { };
+    attrs.axes = {2,3};
+    attrs.mode = "cubic";
+
+    auto raw_images_f32 = std::make_shared<ov::op::v0::Convert>(raw_images_planar, ov::element::f32);
+    auto resized_images_f32 = std::make_shared<ov::op::v0::Interpolate>(raw_images_f32, target_shape, attrs);
+    auto resized_images_m = std::make_shared<ov::op::v1::Subtract>(resized_images_f32, image_mean);
+    auto resized_images_s = std::make_shared<ov::op::v1::Multiply>(resized_images_m, image_scale);
+
+    auto temporal_images = std::make_shared<ov::op::v3::Broadcast>(resized_images_s, broadcast_shape);
+    auto reshaped_8d = std::make_shared<ov::op::v1::Reshape>(temporal_images, temp_shape8d, true);
+    auto transposed_8d = std::make_shared<ov::op::v1::Transpose>(reshaped_8d, 
+        std::make_shared<ov::op::v0::Constant>(element::i32, Shape{8}, std::vector<int32_t>{0,3-1,6-1,4-1,7-1,1,5-1,8-1})
+    );
+    auto reshaped_4d = std::make_shared<ov::op::v1::Reshape>(transposed_8d, temp_shape4d, true);
+    auto transposed_4d = std::make_shared<ov::op::v1::Transpose>(reshaped_4d, 
+        std::make_shared<ov::op::v0::Constant>(element::i32, Shape{4}, std::vector<int32_t>{0,2,1,3})
+    );
+    auto reshaped_to_2d = std::make_shared<ov::op::v1::Reshape>(transposed_4d, last_shape, true);
+
+    auto params_org = model_org->get_parameters();
+    OPENVINO_ASSERT(params_org.size() == 1);
+
+    ov::replace_node(params_org[0], reshaped_to_2d);
+
+    return std::make_shared<ov::Model>(model_org->get_results(), ov::ParameterVector{raw_images, target_shape, image_mean, image_scale, broadcast_shape, temp_shape8d, temp_shape4d, last_shape});
+}
+
+VisionEncoderQwen2VL::VisionEncoderQwen2VL(const std::filesystem::path& model_dir, const std::string& device, const ov::AnyMap properties) : VisionEncoder(model_dir, device, properties) {
+    auto model_org = utils::singleton_core().read_model(model_dir / "openvino_vision_embeddings_model.xml");
+    auto model = patch_preprocess_into_model(model_org);
+    auto compiled_model = utils::singleton_core().compile_model(model, device, properties);
+    ov::genai::utils::print_compiled_model_properties(compiled_model, "VLM vision embeddings model");
+    m_ireq_queue_vision_encoder = std::make_unique<CircularBufferQueue<ov::InferRequest>>(
+        compiled_model.get_property(ov::optimal_number_of_infer_requests),
+        [&compiled_model]() -> ov::InferRequest {
+            return compiled_model.create_infer_request();
+        });
+    m_processor_config = utils::from_config_json_if_exists<ProcessorConfig>(model_dir, "preprocessor_config.json");
```

Comment:
```suggestion
```
`VisionEncoder()` should have already done that

Reply:
Removed, Thanks!